laalaa/apps/advisers/tasks.py

```diff
@@ -82,11 +78,44 @@ def __init__(self, iterable=None, **kwargs):
         super(StrippedDict, self).__init__(iterable, **kwargs)
 
 
-class ImportProcess(Thread):
-    """
-    Loads/Updates data from xls spreadsheet
-    """
-
+class GeocoderTask(Task):
+
+    def __init__(self):
+        self.errors = []
+
+    def run(self, postcodes):
+        tot = len(postcodes)
+        for n, postcode in enumerate(postcodes):
+            pc = ' '.join(postcode[0].encode('utf-8').split())
+            err = None
+            try:
+                point = geocode(pc)
+            except geocoder.PostcodeNotFound:
+                err = 'Failed geocoding postcode: %s' % postcode
+                logging.warn(err)
+            except geocoder.GeocoderError as e:
+                err = 'Failed postcode: "%s" .Error connecting to ' \
+                      'geocoder: %s' % (pc, e)
+                logging.warn(err)
+
+            if err:
+                self.errors.append(err)
```

Comment:
This would be clearer if you encapsulated the error logging in a function, eg:
```python
def run(self, postcodes):
    ...
    def log_error(err):
        logging.warn(err)
        self.errors.append(err)

    for n, postcode ...:
        ...
        except geocoder.PostcodeNotFound:
            log_error('Failed geocoding postcode: {postcode}'.format(postcode=postcode))
            break
```